gammacat/dataset.py

```diff
@@ -0,0 +1,47 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import logging
+from .utils import load_yaml, write_yaml
+from gammapy.catalog.gammacat import GammaCatResource
+
+__all__ = [
+    'DataSet',
+]
+
+log = logging.getLogger(__name__)
+
+class DataSet:
+    """Process a dataset file."""
+    resource_type = 'ds'
+
+    def __init__(self, filename):
```

Comment:
Please use the coding pattern we use everywhere else, where `__init__` takes objects and a separate classmethod `read` does I/O.
https://github.com/gammapy/gamma-cat/blob/ebceb98cbc5beeaf305e5787e203300a2234fd87/gammacat/utils.py#L196